File: js/src/maplestory/api/kms/dto/notice/eventNoticeList.ts
import {
  EventNoticeListBody,
  EventNoticeListItemBody,
} from '../../response/notice/eventNoticeListBody';

/**
 * 진행 중 이벤트 공지 목록
 */
export class EventNoticeListDto {
  /**
   * 공지 목록
   */
  eventNotice: EventNoticeListItemDto[];

  constructor(obj: EventNoticeListBody) {
    const { event_notice } = obj;

    this.eventNotice = event_notice.map(
      (notice) => new EventNoticeListItemDto(notice),
    );
  }
}

/**
 * 진행 중 이벤트 공지
 */
export class EventNoticeListItemDto {
  /**
   * 공지 제목
   */
  title: string;

  /**
   * 공지 링크
   */
  url: string;

  /**
   * 공지 식별자
   */
  noticeId: number;

  /**
   * 공지 등록일
   */
  date: Date;

  /**
   * 이벤트 시작일
   */
  dateEventStart: Date;

  /**
   * 이벤트 종료일
   */
  dateEventEnd: Date;

  constructor(obj: EventNoticeListItemBody) {
    const { title, url, notice_id, date, date_event_start, date_event_end } =
      obj;

    this.title = title;
    this.url = url;
    this.noticeId = notice_id;
    this.date = new Date(date);
    this.dateEventStart = new Date(date_event_start);
    this.dateEventEnd = new Date(date_event_end);
  }
}

File: csharp/MapleStory.OpenAPI/Src/KMS/Dto/Notice/EventNoticeList.cs
using Newtonsoft.Json;

namespace MapleStory.OpenAPI.KMS.DTO
{
    /// <summary>
    /// 진행 중 이벤트 목록
    /// </summary>
    public class EventNoticeListDTO
    {
        /// <summary>
        /// 공지 목록
        /// </summary>
        [JsonProperty("event_notice")]
        public List<EventNoticeListItemDTO> EventNotice { get; set; }
    }

    /// <summary>
    /// 진행 중 이벤트 공지
    /// </summary>
    public class EventNoticeListItemDTO
    {
        /// <summary>
        /// 공지 제목
        /// </summary>
        [JsonProperty("title")]
        public string Title { get; set; }

        /// <summary>
        /// 공지 링크
        /// </summary>
        [JsonProperty("url")]
        public string Url { get; set; }

        /// <summary>
        /// 공지 식별자
        /// </summary>
        [JsonProperty("notice_id")]
        public int NoticeId { get; set; }

        /// <summary>
        /// 공지 등록일
        /// </summary>
        [JsonProperty("date")]
        public DateTimeOffset Date
        {
            get
            {
                return _date.ToOffset(TimeSpan.FromHours(9));
            }
            set
            {
                _date = value;
            }
        }

        private DateTimeOffset _date;

        /// <summary>
        /// 이벤트 시작일
        /// </summary>
        [JsonProperty("date_event_start")]
        public DateTimeOffset DateEventStart
        {
            get
            {
                return _dateEventStart.ToOffset(TimeSpan.FromHours(9));
            }
            set
            {
                _dateEventStart = value;
            }
        }

        private DateTimeOffset _dateEventStart;

        /// <summary>
        /// 이벤트 종료일
        /// </summary>
        [JsonProperty("date_event_end")]
        public DateTimeOffset DateEventEnd
        {
            get
            {
                return _dateEventEnd.ToOffset(TimeSpan.FromHours(9));
            }
            set
            {
                _dateEventEnd = value;
            }
        }

        private DateTimeOffset _dateEventEnd;
    }
}
